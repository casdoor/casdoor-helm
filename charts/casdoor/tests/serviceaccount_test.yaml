suite: test serviceaccount
templates:
- serviceaccount.yaml

tests:
- it: should not render when disabled
  set:
    serviceAccount.create: false
  asserts:
  - hasDocuments:
      count: 0

- it: should render with default values
  set:
    serviceAccount.create: true
  asserts:
  - matchSnapshot: {}

- it: should set a custom service account name
  set:
    serviceAccount.create: true
    serviceAccount.name: my-casdoor-sa
  asserts:
  - matchSnapshot: {}

- it: should add annotations
  set:
    serviceAccount.create: true
    serviceAccount.annotations:
      iam.gke.io/gcp-service-account: casdoor@my-project.iam.gserviceaccount.com
  asserts:
  - matchSnapshot: {}
