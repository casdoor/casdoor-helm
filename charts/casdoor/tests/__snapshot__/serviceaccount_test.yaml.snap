should add annotations:
  1: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      annotations:
        iam.gke.io/gcp-service-account: casdoor@my-project.iam.gserviceaccount.com
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: casdoor
        app.kubernetes.io/version: v1.976.0
        helm.sh/chart: casdoor-helm-charts-v1.976.0
      name: RELEASE-NAME-casdoor
should render with default values:
  1: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: casdoor
        app.kubernetes.io/version: v1.976.0
        helm.sh/chart: casdoor-helm-charts-v1.976.0
      name: RELEASE-NAME-casdoor
should set a custom service account name:
  1: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: casdoor
        app.kubernetes.io/version: v1.976.0
        helm.sh/chart: casdoor-helm-charts-v1.976.0
      name: my-casdoor-sa
